builtin/providers/azure/provider.go

```diff
@@ -0,0 +1,83 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Provider returns a teraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"publish_settings_file": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: parameterDescriptions["publish_settings_file"],
+			},
+			"subscription_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["subscription_id"],
+			},
+			"management_certificate": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["management_certificate"],
+			},
+			"management_url": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["management_url"],
+			},
+		},
+
+		ResourcesMap: map[string]*schema.Resource{
+			"azure_instance":                 resourceAzureInstance(),
+			"azure_hosted_service":           resourceAzureHostedService(),
```

Comment:
Azure has this notion of deployment slots per service.
The most straightforward example I can think of is a web app where you have a 'production' machine serving the current version of your app, and a 'dev' machine on which you add new features etc... In such a case; Azure gives you the huge convenience to 'swap' (it's more or less just switching dns refs) your production machine with the dev one whenever you want to roll out a new version of your app, rollback options etc...

Unfortunately, as of right now, the sdk assumes you wish to use the 'Production' slot all the time; however adding a slot decision option there is trivial.
When such a slot mechanism will be added to the sdk; its use could easily be added to the hostedservice resource as a list of TypeString's and users will have the slots options if they so desire it.